### DISCUSSION TITLE
open/unauth access to api?

### DISCUSSION THREAD

**Discussion by rjadams82** [thomst08/OctoLight#4] [Q&A] [answered]
Does the plugin accept api commands with no api auth? 
Based on my tests I can GET /api/currentuser which returns name:null and I can then GET api/plugin/octolight which toggles the plugin and returns current status.

**Comment by thomst08** [comment]
Hey @rjadams82, that is a good question...
The API part has been there for a while, but your right, there is no check for an API key when querying it...
I will have to look into this and update the code to match with what is recommended with OctoPrint.

Thanks for point that out.

**Comment by thomst08** [comment]
I have tested the plugin and it 100% accepts API calls with no authentication.
I will look into how this could be improved and alter the API function to keep it protected.

**Comment by thomst08** [comment, marked as answer]
I have updated OctoLight to version 0.1.8
This now checks for the "Control" permission when an API call is made.
Please check this has corrected the issue your end and this problem should be resolved.

**Comment by rjadams82** [comment]
Fantastic. I pulled the latest and installed on octoprint 1.9.3
Works as intended. I browse to api/plugin/octolight and received the expected: "error": "You don't have the permission..."

By creating an application key and then simply adding the key in the GET request it works successfully and returns state.
 /api/plugin/octolight?apikey=xxxxxxxxxxxxxxxxxxxx

**Comment by thomst08** [comment]
Excellent, I will close this discussion :)
